src/dataflow/qual/DataFlowTop.java

```diff
@@ -0,0 +1,19 @@
+package dataflow.qual;
+
+import org.checkerframework.framework.qual.DefaultQualifierInHierarchy;
+import org.checkerframework.framework.qual.InvisibleQualifier;
+import org.checkerframework.framework.qual.SubtypeOf;
+import org.checkerframework.framework.qual.TargetLocations;
+import org.checkerframework.framework.qual.TypeUseLocation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+@DefaultQualifierInHierarchy
+@InvisibleQualifier
+@SubtypeOf({})
+@Target({ ElementType.TYPE_USE })
+@TargetLocations({ TypeUseLocation.EXPLICIT_UPPER_BOUND })
+public @interface DataFlowTop {
```

Comment:
Hi @Jianchu, why do you need a separate Qualifier for Dataflow Top (`DataflowTop`) and a separate qualifier for Dataflow Bottom (`DataflowInferenceBottom`)?

Shouldn't `Dataflow` annotation itself can self-contains top (`@Dataflow(roots=Object, names="")`) and bottom (`@Dataflow(roots="", names="")`) with different arguments values as same as Ontology system?

What is the motivation behind these two separate qualifiers `DataflowTop` and `DataflowInferenceBottom`?

It would be very helpful if you could give a brief explanation!

Thanks,